### ISSUE TITLE
mutate() can't cast <Period> to <Period>

### ISSUE THREAD

**Issue by courtiol** [tidyverse/dplyr#4839]
With fresh devel {dplyr} [‘0.8.99.9000’] & {lubridate} [‘1.7.4.9000’] & {vctrs} [‘0.2.99.9005’]
```
lubridate::hm("13:00")
```
works, but

```
tibble(time = "13:00") %>%
  mutate(hm = lubridate::hm(time))
```
does not:
```
> tibble(time = "13:00") %>%
+   mutate(hm = lubridate::hm(time))
Error: Can't cast <Period> to <Period>.
Run `rlang::last_error()` to see where the error occurred.
> rlang::last_error()
Error: Internal error: Trace data is not square.
```

**Comment by romainfrancois** [comment]
I believe this is a `vctrs` issue. cc @lionel- @DavisVaughan 

``` r
vctrs::vec_c(lubridate::hm("13:00"))
#> Error: Can't cast <Period> to <Period>.
vctrs::vec_c(lubridate::hm("13:00"), lubridate::hm("13:00"))
#> Error: No common type for `..1` <Period> and `..2` <Period>.
```

<sup>Created on 2020-02-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

**Comment by lionel-** [comment]
This looks like a lubridate issue?

I wonder if we should add a `base::c()` fallback in `vec_c()` for homogeneously classed inputs?

**Comment by DavisVaughan** [comment]
See also https://github.com/tidyverse/lubridate/issues/721

**Comment by wfjvdham** [comment]
The original problem is solved it seems with versions: {dplyr} [‘0.8.3’] & {lubridate} [‘1.7.4’] & {vctrs} [‘0.2.2’]

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
tibble(time = "13:00") %>%
  mutate(hm = lubridate::hm(time))
#> # A tibble: 1 x 2
#>   time  hm       
#>   <chr> <Period> 
#> 1 13:00 13H 0M 0S
```

<sup>Created on 2020-02-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by courtiol** [comment]
@wfjvdham the problem appear(ed) on versions more recent that those but @lionel- has fixed it (see above) and this issue will be closed once his PR will be validated.
